### ISSUE TITLE
flies.upload: file buffer support

### ISSUE THREAD

**Issue by Nozdi** [os/slacker#115]
`files.upload` does not support file buffers, it accepts only `file_` as a string with the filename.
Use case: create a file buffer with binary data (for example image) and post it to slack without storing.
I tried to post binary file with `content` option, but slack interprets it as a text.

Current implementation:
```
        if file_:
            with open(file_, 'rb') as f:
                return self.post('files.upload', data=data, files={'file': f})
```

Suggested implementation:
```
        if file_:
            if isinstance(file_, str):
                with open(file_, 'rb') as f:
                     return self.post('files.upload', data=data, files={'file': f})
            return self.post('files.upload', data=data, files={'file': file_})           
```

If you think that adding this functionality is a good idea, I'll create a PR :)

**Comment by umitanuki** [comment]
+1

**Comment by os** [comment]
Thanks, @Nozdi. It sounds like a valid request. I'll look into that.

**Comment by os** [comment]
Fixed by #137.
